WordPress/src/main/java/org/wordpress/android/util/config/AppConfig.kt

```diff
@@ -0,0 +1,61 @@
+package org.wordpress.android.util.config
+
+import org.wordpress.android.analytics.AnalyticsTracker.Stat
+import org.wordpress.android.util.analytics.AnalyticsTrackerWrapper
+import org.wordpress.android.util.config.ExperimentConfig.Variant
+import java.lang.IllegalArgumentException
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppConfig
+@Inject constructor(private val remoteConfig: RemoteConfig, private val analyticsTracker: AnalyticsTrackerWrapper) {
+    /**
+     * We need to keep the value of an already loaded feature flag to make sure the value is not changed while using the app.
+     * We should only reload the flags when the application is created.
+     */
+    private val enabledFeatures = mutableMapOf<FeatureConfig, Boolean>()
+    private val experimentValues = mutableMapOf<ExperimentConfig, String>()
+
+    /**
+     * This method initialized the config and triggers refresh of remote configuration.
+     */
+    fun refresh() {
+        remoteConfig.refresh()
+    }
+
+    /**
+     * Get the enabled state of a feature flag. If the flag is enabled in the BuildConfig file, it overrides the
+     * remote value. The correct approach is to disable a feature flag for a release version and only enable it remotely.
+     * Once the feature is ready to be fully released, we can enable the BuildConfig value.
+     * @param feature feature which we're checking remotely
+     */
+    fun isEnabled(feature: FeatureConfig): Boolean {
+        if (feature.remoteField == null) {
+            return feature.buildConfigValue
+        }
+        return enabledFeatures.getOrPut(feature) {
```

Comment:
I believe this doesn't work as intended. 

1. Inject `TenorFeatureConfig` into a fragment
2. Dagger creates an instance of `TenorFeatureConfig`
3. Invoke `isEnabled` -> the config isn't found, so it's created
4. Inject `TenorFeatureConfig` into a different fragment
5. Dagger creates a NEW instance of `TenorFeatureConfig`
6. Invoke `isEnabled` -> the config isn't found, so it's created

I believe we need to store a unique value - eg `class`/FeatureConfig.remoteField into the map. Wdyt?

Reply:
ah, you're right 👍 , we need to use the field.